swap_meet/vendor.py

```diff
@@ -0,0 +1,123 @@
+class Vendor:    
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+    
+    def add(self, item):
+        """
+        Adds item to the inventory 
+        Args:
+            item (Item): inventory list
+        Returns:
+            item: added item
+        """        
+        (self.inventory).append(item) 
+        return item
+
+    def remove(self, item): 
+        """
+        Removes the matching item from the inventory
+        Args:
+            item (Item): inventory list
+        Returns:
+            item: removed item
+        """        
+        item_index = None
+        for index in range(len(self.inventory)):
+            if item == self.inventory[index]:
+                return self.inventory.pop(index)
+        return False
+
+    def get_by_category(self, category):
+        """
+        Returns a list of Items in the inventory with that category  
+        Args:
+            category (string): attribute of class Item
+        Returns:
+            list: list of Items in the inventory with the category
+        """        
+        return [item for item in self.inventory if item.category == category] 
+    
+    def swap_items(self, friend_vendor, my_item, their_item):
+        """
+        It removes my_item from this Vendor's inventory, and adds it 
+        to the friend_vendor inventory. It removes the their_item
+        from the friend_vendor inventory, and adds it to this 
+        Vendor's inventory
+        Args:
+            friend_vendor (Vendor): an instance of another Vendor
+            my_item (Item): an instance of this Vendor's Item 
+            their_item (Item): an instance of friend_vendor (Vendor) 
+            Item 
+        Returns:
+            Boolean: True if swap is successful, False otherwise.
+        """        
+        friend_list = friend_vendor.inventory  
+        my_list = self.inventory  
+        if their_item in friend_list and my_item in my_list \
+            and their_item != None and my_item != None and my_list != []\
+            and friend_list !=[]:
```

Comment:
I love how thorough this conditional is! This covers edge cases the tests don't by making sure the items themselves aren't `None`.
That said, I want to point out that `my_list != [] and friend_list !=[]` are both not really necessary because the checks for `their_item in friend_list and my_item in my_list` will return false if either of the lists are empty